src/routers/controllers/addUserController.ts

```diff
@@ -23,7 +25,9 @@ export const addUserControllerGet = async (req: Request, res: Response): Promise
         loggedInUserRole,
         templateName: constants.ADD_USER_PAGE
     };
-    clearFormSessionValues(req, constants.DETAILS_OF_USER_TO_ADD);
+    if (!referrer?.includes(hrefB)) {
+        clearFormSessionValues(req, constants.DETAILS_OF_USER_TO_ADD);
```

Comment:
When we switch languages it might be better to just not include the cf=true param in the url, which is probably causing the clearFormSessionValues to be triggered rather than using the referrer. Have you tried that already?